dot/sync/disjoint_block_set_test.go

```diff
@@ -0,0 +1,313 @@
+			t.Parallel()
+			err := tt.disjointBlockSet.addBlock(tt.block)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_disjointBlockSet_addHeader(t *testing.T) {
+	t.Parallel()
+
+	mockTime := func() time.Time { return time.Unix(1001, 0) }
+
+	tests := []struct {
+		name             string
+		disjointBlockSet disjointBlockSet
+		header           *types.Header
+		err              error
+	}{
+		{
+			name: "add header beyond capactiy",
+			header: &types.Header{
+				Number: 1,
+			},
+			err: errors.New("cannot add block; set is at capacity"),
+		},
+		{
+			name: "add header",
+			disjointBlockSet: disjointBlockSet{
+				blocks:           make(map[common.Hash]*pendingBlock),
+				limit:            1,
+				timeNow:          mockTime,
+				parentToChildren: make(map[common.Hash]map[common.Hash]struct{}),
+			},
+			header: &types.Header{
+				Number: 1,
+			},
+		},
+	}
+	for _, tt := range tests { //nolint:govet
+		tt := tt //nolint:govet
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			err := tt.disjointBlockSet.addHeader(tt.header)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_disjointBlockSet_clearBlocks(t *testing.T) {
+	t.Parallel()
+
+	testBlock := &pendingBlock{
+		clearAt: time.Unix(1000, 0),
+		hash:    common.Hash{1},
+	}
+	testBlock2 := &pendingBlock{
+		clearAt: time.Unix(1002, 0),
+		hash:    common.Hash{2},
+	}
+	mockTime := func() time.Time { return time.Unix(1001, 0) }
+
+	tests := []struct {
+		name             string
+		disjointBlockSet disjointBlockSet
+		remaining        int
+	}{
+		{
+			name: "base case",
+			disjointBlockSet: disjointBlockSet{
+				limit: 0,
+				blocks: map[common.Hash]*pendingBlock{
+					{1}: testBlock,
+				},
+				timeNow: mockTime,
+			},
+			remaining: 0,
+		},
+		{
+			name: "remove clear one block",
+			disjointBlockSet: disjointBlockSet{
+				limit: 0,
+				blocks: map[common.Hash]*pendingBlock{
+					{1}: testBlock,
+					{2}: testBlock2,
+				},
+				timeNow: mockTime,
+			},
+			remaining: 1,
+		},
+	}
+	for _, tt := range tests { //nolint:govet
+		tt := tt //nolint:govet
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			tt.disjointBlockSet.clearBlocks()
+			assert.Equal(t, tt.remaining, len(tt.disjointBlockSet.blocks))
+		})
+	}
+}
+
+func Test_disjointBlockSet_getBlocks(t *testing.T) {
+	t.Parallel()
+
+	testBlock := &pendingBlock{}
+
+	tests := []struct {
+		name             string
+		disjointBlockSet disjointBlockSet
+		want             []*pendingBlock
+	}{
+		{
+			name: "no blocks",
+			want: []*pendingBlock{},
+		},
+		{
+			name: "base case",
+			disjointBlockSet: disjointBlockSet{
+				blocks: map[common.Hash]*pendingBlock{
+					{}: testBlock,
+				},
+			},
+			want: []*pendingBlock{testBlock},
+		},
+	}
+	for _, tt := range tests { //nolint:govet
+		tt := tt //nolint:govet
```

Comment:
nit change your test case to use a `*disjointBlockSet` instead of `disjointBlockSet` and remove those `//nolint:govet`. That way the mutex lock isn't 'copied' and the linter won't complain.

Reply:
Updated.